### PR TITLE
Now it is possible to pick a file in Downloads folder.

### LINKED ISSUE #791: File selection does not work properly
When I select a PDF from the downloads dir of the phone for the remove password feature, the file is not selected and the wrong path is shown, if I put he file in a subdir it works again
Here is a video reproducing the issue 
https://drive.google.com/file/d/10irvh4LMb5tPuex5Fu8lzptTd3Y51_tA/view?usp=drivesdk

### PR DESCRIPTION
**Pull request by hector2**
# Description

Now it is possible to pick a file in Downloads folder. But this change brings some discussion:
- Kudos to https://github.com/HBiSoft/PickiT. Basically I used the classes of his library, but at the moment we can't just add the library as a dependency, because it uses androidX and this project doesn't.  If we migrate to androidX we could just add the dependency, remove the package "pickit" and use the classes of the library.
- I added the package "pickit" to exclusions in checkstyle because the idea is to migrate to androidX and add the dependency to PickiT. If you want me to remove that exclusion and edit the classes tell me 💃 

Fixes #791

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.
- [x] `./gradlew assembleDebug assembleRelease`
- [x] `./gradlew checkstyle`

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: config/checkstyle.gradle [hunk 1]
```diff
@@ -12,6 +12,7 @@ check.dependsOn 'checkstyle'
 task checkstyle(type: Checkstyle, group: 'Verification', description: 'Runs code style checks') {
     configFile file("$qualityConfigDir/checkstyle_config.xml")
     source 'src'
+    exclude '**/swati4star/createpdf/pickit/**'
```
Comment (Swati4star) [inline config/checkstyle.gradle]: why this?

**Comment by Swati4star** [comment]
@hector2 Can this issue only be solved by integration of above said library?
I think, we should be handle these in code.
I am not in favor of adding a new library (increasing app size) just for this purpose.

**Comment by Swati4star** [comment]
@hector2 updates?

**Comment by hector2** [comment]
Hello @Swati4star yeah you are right I'll try to handle this in code only. I'm on it 😄

**Comment by hector2** [comment]
@Swati4star hey I updated the branch, what do you think?

File: app/src/main/java/swati4star/createpdf/util/RealPathUtil.java [hunk 2]
```diff
@@ -137,4 +208,13 @@ private static boolean isDriveFile(Uri uri) {
     private static boolean isDownloadsDocument(Uri uri) {
         return "com.android.providers.downloads.documents".equals(uri.getAuthority());
     }
+
+    /**
+     * @param uri The Uri to check
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/util/RealPathUtil.java]: follow standard javadoc style comments

File: build.gradle [hunk 3]
```diff
@@ -9,7 +9,7 @@ buildscript {
         google()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.4.2'
+        classpath 'com.android.tools.build:gradle:3.5.1'
```
Comment (Swati4star) [inline build.gradle]: revert this change.

File: gradle/wrapper/gradle-wrapper.properties [hunk 4]
```diff
@@ -1,6 +1,6 @@
-#Mon Aug 12 00:16:23 PDT 2019
+#Sat Oct 05 20:58:28 CEST 2019
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-5.1.1-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-5.4.1-all.zip
```
Comment (Swati4star) [inline gradle/wrapper/gradle-wrapper.properties]: revert this change

File: app/src/main/java/swati4star/createpdf/util/RealPathUtil.java [hunk 5]
```diff
@@ -30,53 +32,122 @@ public static String getRealPath(Context context, Uri fileUri) {
...
+            return sub3 + "/" + sub4 + "/" + sub5 + "/";
+        } else if (sub3.equals("Download")) {
+            return sub4 + "/" + sub5 + "/";
+        } else if (sub4.equals("Download")) {
+            return sub5 + "/";
         } else {
-            StringBuilder pathbuilder = StringUtils.trimExternal(uri.getPath().substring(1));
-            pathbuilder.insert(0, Environment.getExternalStorageDirectory() + "/");
-            path =  pathbuilder.toString();
+            return "";
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/util/RealPathUtil.java]: return null

File: app/src/main/java/swati4star/createpdf/util/RealPathUtil.java [hunk 6]
```diff
@@ -30,53 +32,122 @@ public static String getRealPath(Context context, Uri fileUri) {
...
+    private static String getSubFolders(Uri uri) {
+        String replaceChars = String.valueOf(uri).replace("%2F", "/")
+                .replace("%20", " ").replace("%3A", ":");
+        String[] bits = replaceChars.split("/");
+        String sub5 = bits[bits.length - 2];
+        String sub4 = bits[bits.length - 3];
+        String sub3 = bits[bits.length - 4];
+        String sub2 = bits[bits.length - 5];
+        String sub1 = bits[bits.length - 6];
+        if (sub1.equals("Download")) {
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/util/RealPathUtil.java]: It isnt exactly clear what are you trying to do here?
Add comments.

File: app/src/main/java/swati4star/createpdf/util/RealPathUtil.java [hunk 7]
```diff
@@ -30,53 +32,122 @@ public static String getRealPath(Context context, Uri fileUri) {
...
+    /**
+     * Get all the subfolders from an Uri.
+     *
+     * @param uri The uri
+     * @return A string containing all the subfolders that point to the final file path
+     */
+    private static String getSubFolders(Uri uri) {
+        String replaceChars = String.valueOf(uri).replace("%2F", "/")
+                .replace("%20", " ").replace("%3A", ":");
+        String[] bits = replaceChars.split("/");
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/util/RealPathUtil.java]: change 'bits' to components

File: app/src/main/java/swati4star/createpdf/util/RealPathUtil.java [hunk 8]
```diff
@@ -30,53 +32,122 @@ public static String getRealPath(Context context, Uri fileUri) {
...
-            // check if path contains :
-            path = path.replaceAll("[:]", "/");
+    /**
+     * Get the file path (without subfolders if any)
+     *
+     * @param context The context
+     * @param uri     The uri to query
+     * @return The file path
+     */
+    @SuppressWarnings("TryFinallyCanBeTryWithResources")
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/util/RealPathUtil.java]: is it needed?

File: build.gradle (hunk 3 above)
Comment (hector2) [reply build.gradle]: done

File: gradle/wrapper/gradle-wrapper.properties (hunk 4 above)
Comment (hector2) [reply gradle/wrapper/gradle-wrapper.properties]: done

File: app/src/main/java/swati4star/createpdf/util/RealPathUtil.java (hunk 8 above)
Comment (hector2) [reply app/src/main/java/swati4star/createpdf/util/RealPathUtil.java]: No, I can use try-with-resources

File: app/src/main/java/swati4star/createpdf/util/RealPathUtil.java (hunk 5 above)
Comment (hector2) [reply app/src/main/java/swati4star/createpdf/util/RealPathUtil.java]: done

File: app/src/main/java/swati4star/createpdf/util/RealPathUtil.java (hunk 6 above)
Comment (hector2) [reply app/src/main/java/swati4star/createpdf/util/RealPathUtil.java]: is it clear now? although I have to test it with longer paths and subfolders.

File: app/src/main/java/swati4star/createpdf/util/RealPathUtil.java (hunk 7 above)
Comment (hector2) [reply app/src/main/java/swati4star/createpdf/util/RealPathUtil.java]: done
